doc/fast-reboot/Fast-reboot_Flow_Improvements_HLD.md

```diff
@@ -0,0 +1,158 @@
+# Fast-reboot Flow Improvements HLD
+
+# Table of Contents
+- [Fast-reboot Flow Improvements HLD](#fast-reboot-flow-improvements-hld)
+- [Table of Contents](#table-of-contents)
+- [List of Figures](#list-of-figures)
+- [1 Overview](#1-overview)
+- [2 Functional Requirements](#2-functional-requirements)
+- [3 Use Cases](#3-use-cases)
+  - [3.1 In-Service restart and upgrade](#31-in-service-restart-and-upgrade)
+- [4 Reconciliation at syncd](#4-reconciliation-at-syncd)
+  - [4.1 Orchagent Point Of View](#41-orchagent-point-of-view)
+  - [4.2 Syncd Point Of View - INIT/APPLY view framework](#42-syncd-point-of-view---initapply-view-framework)
+  - [4.3 neighsyncd Point Of View](#43-neighsyncd-point-of-view)
+  - [4.4 fpmsyncd Point Of View](#44-fpmsyncd-point-of-view)
+  - [4.5 Reboot finalizer](#45-reboot-finalizer)
+- [5 SONiC Application Extension Infrastructre Integration](#5-sonic-application-extension-infrastructre-integration)
+
+# List of Figures
+* [Syncd and Orchagent](#42-syncd-point-of-view---initapply-view-framework)
+* [Neighbors](#43-neighsyncd-point-of-view)
+* [BGP](#44-fpmsyncd-point-of-view)
+* [Finalizer](#45-reboot-finalizer)
+
+# 1 Overview
+
+The goal of SONiC fast-reboot is to be able to restart and upgrade SONiC software with a data plane disruption less than 30 seconds and control plane less than 90 seconds.
+With current implementation there is no indication of the fast-reboot status, meaning we don't have a way to determine if the flow has finished or not.
+Some feature flows in SONiC are delayed with a timer to keep the CPU dedicated to the fast-reboot init flow for best perforamnce, like enablement of flex counters.
+In order to have such indicator, re-use of the fastfast-reboot infrastructure can be used.
```

Comment:
Minor: can we just call it `warm-reboot` instead of `fastfast-reboot`. Warm-reboot is generic.